### PR TITLE
chore: use cross-env

### PR DESCRIPTION
**Pull request by h-a-n-a**
## Summary

closes https://github.com/web-infra-dev/rspack/issues/4128

### CODE REVIEW

**Review by hardfist — APPROVED** [review]

**Comment by hardfist** [comment]
!canary

File: package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
     "prepare": "is-ci || husky install",
     "pnpm:devPreinstall": "is-ci || node ./scripts/doctor.js",
     "test:example": "pnpm --filter \"example-*\" build",
-    "test:unit": "NO_COLOR=1 pnpm --filter \"@rspack/*\" test",
+    "test:unit": "cross-env NO_COLOR=1 pnpm --filter \"@rspack/*\" test",
```
Comment (ahabhgk) [inline package.json:27]: Could you move `NO_COLOR=1` to `packages/rspack/package.json`, some people (me) will use that command to run test

**Review by ahabhgk — APPROVED** [review]

**Comment by h-a-n-a** [comment]
!eco-ci

**Comment by h-a-n-a** [comment]
!canary

**Comment by hardfist** [comment]
decide to move NO_COLOR to jest config

**Comment by hardfist** [comment]
> decide to move NO_COLOR to jest config

turns out addon woudn't read process.env

**Comment by h-a-n-a** [comment]
Let's close this as normally we don't snapshot color(maybe cross platform) https://github.com/web-infra-dev/rspack/issues/4124

**Comment by hardfist** [comment]
i think the ideal solution is swc-loader support noColor options so test can control it progmatically and avoid env limitation

**Pull request merged**